astronomer/providers/snowflake/operators/snowflake.py

```diff
@@ -56,6 +56,19 @@ def execute(self, context: Context) -> None:
         By deferring the SnowflakeTrigger class pass along with query ids.
         """
         self.log.info("Executing: %s", self.sql)
+
+        default_query_tag = (
+            f"airflow_openlineagens_{self.task_id}_{self.dag.dag_id}_{context['ti'].try_number}"
+        )
+        query_tag = (
+            self.parameters.get("query_tag", default_query_tag) if self.parameters else default_query_tag
+        )
+        session_query_tag = f"ALTER SESSION SET query_tag = '{query_tag}';"
+        if isinstance(self.sql, str):
```

Comment:
do we also need a unit test for this block?

Reply:
Yes

Reply:
could you point me to the  test for this new functionality